internal/server/iigointernal/orchestration.go

```diff
@@ -143,27 +172,54 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	presidentMonitored := monitoring.monitorRole(iigoClients[g.SpeakerID])
 
 	// 3 Speaker actions
-	legislativeBranch.setRuleToVote(ruleToVote)
-	voteCalled := legislativeBranch.setVotingResult(aliveClientIds)
-	legislativeBranch.announceVotingResult()
+
+	//TODO:- shouldn't updateRules be called here?
+	//TODO: Check if outer if statement is needed
+	var voteCalled bool = false
+	if ruleToVoteReturn.ActionTaken && ruleToVoteReturn.ContentType == shared.PresidentRuleProposal {
+		insufficientBudget := legislativeBranch.setRuleToVote(ruleToVoteReturn.ProposedRule)
+		if insufficientBudget != nil {
+			return false, "Common pool resources insufficient for legislativeBranch setRuleToVote"
+		}
+		insufficientBudget = legislativeBranch.setVotingResult(aliveClientIds)
+		if insufficientBudget != nil {
+			return false, "Common pool resources insufficient for legislativeBranch setVotingResult"
+		}
+		insufficientBudget = legislativeBranch.announceVotingResult()
+		if insufficientBudget != nil {
+			return false, "Common pool resources insufficient for legislativeBranch announceVotingResult"
+		}
+		voteCalled = true
+	}
 
 	variablesToCache = []rules.VariableFieldName{rules.RuleSelected, rules.VoteCalled}
 	valuesToCache = [][]float64{{boolToFloat(ruleSelected)}, {boolToFloat(voteCalled)}}
 	monitoring.addToCache(g.SpeakerID, variablesToCache, valuesToCache)
 
 	speakerMonitored := monitoring.monitorRole(iigoClients[g.JudgeID])
 
+	// TODO:- at the moment, these are action (and cost resources) but should they?
+	var appointJudgeError, appointSpeakerError, appointPresidentError error
```

Comment:
Why not drop these and use the := instantiation?

Reply:
Because it doesn't let you update the game state in that way. If you add := on appointNextJudge for example you get an error at g.JudgeID

Reply:
Oh right I understand, soz didn't notice